src/Microsoft.DotNet.Darc/Darc/Operations/VirtualMonoRepo/VmrDiffOperation.cs

```diff
@@ -0,0 +1,343 @@
+using Microsoft.DotNet.Darc.Helpers;
+using Microsoft.DotNet.Darc.Options.VirtualMonoRepo;
+using Microsoft.DotNet.DarcLib;
+using Microsoft.DotNet.DarcLib.Helpers;
+using Microsoft.DotNet.DarcLib.VirtualMonoRepo;
+
+#nullable enable
+namespace Microsoft.DotNet.Darc.Operations.VirtualMonoRepo;
+
+internal class VmrDiffOperation(
+    VmrDiffOptions options,
+    IProcessManager processManager,
+    IFileSystem fileSystem,
+    IGitRepoFactory gitRepoFactory,
+    IVersionDetailsParser versionDetailsParser,
+    IVmrPatchHandler patchHandler,
+    ISourceMappingParser sourceMappingParser) : Operation
+{
+    private const string GitDirectory = ".git";
+    private readonly static string GitSparseCheckoutFile = Path.Combine(GitDirectory, "info", "sparse-checkout");
+    private const string HttpsPrefix = "https://";
+
+    public override async Task<int> ExecuteAsync()
+    {
+        (DiffRepo repo1, DiffRepo repo2) = await ParseInput();
+
+        string tmpPath = Path.Combine(Path.GetTempPath(), Path.GetRandomFileName());
+        try
+        {
+            string tmpProductRepo, tmpVmrProductRepo;
+            fileSystem.CreateDirectory(tmpPath);
+            string mapping;
+            if (repo1.IsVmr)
+            {
+                // Think we don't have to actually clone the product repo, we can just add github as a remote..
+                tmpProductRepo = await PrepareRepo(repo2, tmpPath);
+                mapping = versionDetailsParser.ParseVersionDetailsFile(Path.Combine(tmpProductRepo, "eng", "Version.Details.xml")).Source?.Mapping
+                    ?? Path.GetFileName(tmpProductRepo);
+                tmpVmrProductRepo = await PrepareVmr(repo1, tmpPath, mapping);
+            }
+            else
+            {
+                tmpProductRepo = await PrepareRepo(repo1, tmpPath);
+                mapping = versionDetailsParser.ParseVersionDetailsFile(Path.Combine(tmpProductRepo, "eng", "Version.Details.xml")).Source?.Mapping
+                    ?? Path.GetFileName(tmpProductRepo);
+                tmpVmrProductRepo = await PrepareVmr(repo2, tmpPath, mapping);
+            }
+            
+            await AddRemoteAndGenerateDiff(tmpProductRepo, tmpVmrProductRepo, repo2.Branch, await GetDiffFilters(mapping));
+        }
+        finally
+        {
+            if (fileSystem.DirectoryExists(tmpPath))
+            {
+                var gitFiles = Directory.GetDirectories(tmpPath, GitDirectory, SearchOption.AllDirectories)
+                    .Select(gitDir => Path.Combine(gitDir, "objects"))
+                    .SelectMany(q => Directory.GetFiles(q, "*", SearchOption.AllDirectories));
+                foreach (var gitFile in gitFiles)
+                {
+                    var fileInfo = new FileInfo(gitFile);
+                    fileInfo.Attributes = FileAttributes.Normal;
+                    fileInfo.IsReadOnly = false;
+                }
+                fileSystem.DeleteDirectory(tmpPath, true);
```

Comment:
Why are we deleting the git files like this?

Reply:
We're not deleting them, we're just doing this so they can be deleted. For whatever reason, if we don't do this, the app can't delete them

Reply:
Ah, I see